numpy/core/src/common/simd/avx512/arithmetic.h

```diff
@@ -181,4 +204,10 @@ NPY_FINLINE npy_uint32 npyv_sum_u32(npyv_u32 a)
     }
 #endif
 
+NPY_FINLINE npy_uint32 npyv_sum_u16(__m512i a)
+{
+    npyv_u32x2 res = npyv_expand_u32_u16(a);
+    return (unsigned)npyv_sum_u32(_mm512_add_epi32(res.val[0], res.val[1]));
+}
```

Comment:
```suggestion
NPY_FINLINE npy_uint32 npyv_sum_u16(npyv_u16 a)
{
    const npyv_u16 even_mask = _mm512_set1_epi32(0x0000FFFF);
    __m512i even = _mm512_and_si512(a, even_mask);
    __m512i odd  = _mm512_srli_epi32(a, 16);
    __m512i ff   = _mm512_add_epi32(even, odd);
    return npyv_sum_u32(ff);
}
```
same as the `AVX2` one.